src/runtime/prte_globals.c

```diff
@@ -704,6 +705,11 @@ static void prte_job_destruct(prte_job_t* job)
 
     PRTE_DESTRUCT(&job->launch_msg);
 
+    /* Clear the child list before destroying the list */
```

Comment:
This logic works in prte_finalize since we are in cleanup and don't care much about what happens next. Here, I'm wondering if this is masking some other bug and that this case should never happen in a properly functioning system. The potential problem I see is that you are destroying a parent job before destroying its children. Are there any implications to that? As a minimum, I think there is a lingering prte_job_data structure that then gets claned up in prte_finalize.

Reply:
That's an interesting idea. I'd have to look in the code and ask the community.

An alternative way to clean this up would be to do a depth-first cleanup of the children jobs. This would guarentee that the children are always cleaned up before their parent.

I suppose we could also destruct the jobs in reverse order (highest job number to lowest). Since the jobs are given out in sequential order that should mean that children are cleaned up before the parents. That does assume that children alreays have a higher job id than their parent. Which I would need to think about a bit.

My feeling is that since we are in finalize we are ok, but this is definitely somethign we should check.